src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -590,6 +592,15 @@ public List<Dataset> getSourceDatasetsFromSnapshotRequest(
         .collect(Collectors.toList());
   }
 
+  public PatchAuthDomainResponseModel patchSnapshotAuthDomain(
+      AuthenticatedUserRequest userReq, UUID snapshotId, List<String> userGroups) {
+    String description = "Patch auth domain for snapshot " + snapshotId;
+    return jobService
+        .newJob(description, SnapshotPatchAuthDomainFlight.class, userGroups, userReq)
+        .addParameter(JobMapKeys.SNAPSHOT_ID.getKeyName(), snapshotId.toString())
+        .submitAndWait(PatchAuthDomainResponseModel.class);
```

Comment:
>There is a short but noticeable delay, so I'm interested in hearing what others think about keeping it this way vs. returning a job id.

Thanks for flagging this!  A few follow-on questions:

Do you have a sense of how this endpoint will be called?  
Down the road, might there be a benefit to allowing a snapshot's auth domains to be specified on creation?
Roughly how long a delay have you seen?

My instinct is that a short but noticeable delay is okay (at most a few seconds), since this endpoint would be called at most once per snapshot.

Reply:
Makes good sense (not expanding snapshot creation in this PR).

I had asked because I wasn't sure if we thought that this endpoint would be called via TDR UI, which already has capability for polling async jobs to completion.

One thing that came to mind: under the covers I know that we're communicating with Sam within the called flight, and sometimes when Sam's acting up we find ourselves in its retry loop, which can extend the total runtime as it backs off and waits.  I don't think this is super likely to happen though.